feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+from feature_engine.discretisation import (
+    EqualFrequencyDiscretiser,
+    EqualWidthDiscretiser,
+)
+from feature_engine.selection.base_selector import BaseSelector
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class DropHighPSIFeatures(BaseSelector):
+    """
+    DropHighPSIFeatures drops features with a Population Stability Index (PSI) above
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
+
+    In fields like Credit Risk Modelling, the elimination of features with high PSI
+    is frequent and usually required by the Regulator.
+
+    The calculation of the PSI requires to compare two distributions.
+    In DropHighPSIFeatures the approach implemented is the following:
+
+    - The dataframe is split in two parts and the PSI calculations are performed by
+    comparing the distribution of the features in the two parts.
+
+    - The user defines the columns of the dataframe according to which the dataframe
+    will de split. The column can contain numbers, dates and strings.
+
+    - The user defines a split fraction. This is used to define the sizes of the two
+    dataframe. Is the split fraction is 50%, the two dataframe will have about the same
+    size. It is not guaranteed that the size will exactly match because the observations
+    are grouped by their label (i.e. their values in the split columns) so all
+    observations with the same label will automatically belong to the same dataframe.
+
+    - If the option split_distinct_value is activated, the number of distinct values
```

Comment:
I guess this would only apply if the column is categorical, but then, would one really use a categorical column to split the dataframe? thinking of finance?